yucca/functional/evaluation/evaluate_folder.py

```diff
@@ -13,195 +13,228 @@
-            tn = np.sum(cmat) - tp - fp - fn  # often a redundant and meaningless metric
-            for k, v in metrics.items():
-                labeldict[k] = round(v(tp, fp, tn, fn), 4)
-                meandict[str(region_name)][k].append(labeldict[k])
-
-            if obj_metrics:
-                raise NotImplementedError
-                # now for the object metrics
-                # obj_labeldict = get_obj_stats_for_label(gt, pred, label, spacing=spacing, as_binary=as_binary)
-                # for k, v in obj_labeldict.items():
-                #    labeldict[k] = round(v, 4)
-                #    meandict[str(label)][k].append(labeldict[k])
-
-            if surface_metrics:
-                if label == 0:
-                    surface_labeldict = get_surface_metrics_for_label(
-                        gt[label], pred[label], 0, spacing=spacing, tol=surface_tol, as_binary=as_binary
-                    )
-                else:
-                    surface_labeldict = get_surface_metrics_for_label(
-                        gt[label - 1], pred[label - 1], label, spacing=spacing, tol=surface_tol, as_binary=as_binary
-                    )
-                for k, v in surface_labeldict.items():
-                    labeldict[k] = round(v, 4)
-                    meandict[str(region_name)][k].append(labeldict[k])
-            casedict[str(region_name)] = labeldict
-        casedict["Prediction:"] = predpath
-        casedict["Ground Truth:"] = gtpath
-
-        resultdict[case] = casedict
-        del pred, gt, cmat
-
-    for label in regions.keys():
-        meandict[str(label)] = {
-            k: round(np.nanmean(v), 4) if not np.all(np.isnan(v)) else 0 for k, v in meandict[str(label)].items()
+    for label in labels:
+        mean_dict[str(label)] = {
+            k: round(np.nanmean(v), 4) if not np.all(np.isnan(v)) else 0 for k, v in mean_dict[str(label)].items()
         }
-    resultdict["mean"] = meandict
-    return resultdict
+    result_dict["mean"] = mean_dict
 
+    return result_dict
 
-def evaluate_folder_segm(
-    labels,
-    metrics,
-    subjects,
-    folder_with_predictions,
-    folder_with_ground_truth,
-    as_binary: Optional[bool] = False,
-    obj_metrics: Optional[bool] = False,
-    surface_metrics: Optional[bool] = False,
-    surface_tol=1,
+
+def evaluate_multilabel_case_segm(
+    case: str,
+    metrics: dict,
+    folder_with_predictions: str,
+    folder_with_ground_truth: str,
+    labels: dict,
+    labels_from_regions: np.array,
+    as_binary: bool,
+    obj_metrics: bool,
+    surface_metrics: bool,
+    surface_tol: int,
+    regions: dict,
 ):
-    logging.info(f"segmentation evaluation with labels: {labels}")
+    assert regions is not None
+
+    case_dict = {}
+    predpath = join(folder_with_predictions, case)
+    gtpath = join(folder_with_ground_truth, case)
+    case_dict["Prediction:"] = predpath
+    case_dict["Ground Truth:"] = gtpath
+
+    pred = nib.load(predpath)
```

Comment:
Very nice, I think if we just added functionality to load any file type here rather than only nifti, then we could make these functions work on any data 
Then it should just return spacing = [1., 1., 1.,] for 3D or [1., 1.,] for 2D data, and then regions would also work as intended. However I understand if you don't wanna do that right now

Reply:
I will leave that for future work :-)